### PR TITLE
Complete Re-work of Chef-Bareos Cookbook

### PR DESCRIPTION
**Pull request by GitBytes**
@sitle 

I have been working on this cookbook over the past nine months or so. It is now to the point (other than a couple of README bits I still need to fill in) where I feel it is worth having you take a look to see if you would like to get this design of the cookbook up and going on the supermarket. I am happy to finish the remaining README bits completed before moving forward but I wanted to get this to you to look over if you have time.

I am happy to have a discussion on things as needed. This is not a small merge but it is much more than it was in the past I think. Let me know if you get time to look through it. I am currently using this in production, pulling directly from our repo for now. If we can get this on the supermarket in the next couple of weeks great (assuming the README gets completed) but overall I am in no rush. Thanks!

(FYI Example Hashes are included in the attributes and tests work via kitchen. ChefSpec tests need a bit of work yet to complete, I also just want more spec tests in general since they are lacking at the moment but I haven't had enough time to retrofit them yet.)

BR
-Ian

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,11 +1,39 @@
-Bareos Cookbook CHANGELOG
-==========================
+CHANGELOG
+=========
+Chef-Bareos Cookbook
+--------------------
 
-This file is used to list changes made in each version of the bareos cookbook.
+2.2.13
```
Comment (GitBytes) [inline CHANGELOG.md:21]: I need to add some kind of entry for the current version before this should be merged. I'll give an explanation as to why I did the rework and going forward what the plan may entail...

**Comment by treenerd** [comment]
Wow, thats a lot of work! Amazing. Thank you very much.

**Comment by sitle** [comment]
It's definitively awesome, thx for your work :p 

I need to test a little bit more for myself, but i'm fine to merge it and update the supermarket. What do you think ?

Best regard.

**Comment by GitBytes** [comment]
@sitle @treenerd 

Wow! I am really pleased you guys like it so far! I am itchy to pull the trigger to update the supermarket but I think I would still like to hold off until I get the docs more complete-ish. Thank you so much for the quick offer though @sitle. As soon as I can find a tad more time this week I will try and polish the docs up for a supermarket release, including the README and the CHANGELOG (summarized). I am 50/50 whether to merge this request anyways to the develop branch then still finish docs before going to market (master). I would say if you guys would like to merge #15 but not update the market (master branch) until I can get a docs merge request in the works I think that would work. @treenerd Feel free to chip in there if you would like. Once you both have had enough time to test things and see if it is ok as is, feel free to merge to develop for now. Great to hear from you guys so quickly, cheers!!

-Ian

**Comment by GitBytes** [comment]
@sitle @treenerd 
Ok guys.. It should be ready given your tests work out ok. Feel free to merge to both Develop and master and push to the supermarket when ready and Travis-CI tests pass. Cheers!!

-Ian

**Comment by sitle** [comment]
@GitBytes All done. Thx again for your work.

Best regard.

**Comment by GitBytes** [comment]
@sitle Happy to help! I have a tiny bit of cleanup for the README again but I'll wait to push that out until I have something else to get out there.. It is a Supermarket README cosmetic update so low priority.. Formatting got a little wonky I noticed after being posted.. Oh well. Thanks again for getting it up on the Supermarket nonetheless! :) Have a great weekend, cheers!

-Ian

**Pull request merged**